Pages/Index.cshtml.cs

```diff
@@ -1,100 +1,87 @@
 
-namespace TwilioSMSScheduler.Pages;
+namespace TwilioSmsScheduler.Pages;
 
 public class IndexModel : PageModel
 {
-    private readonly ILogger<IndexModel> _logger;
-    private readonly IConfiguration _configuration;
-    private string clientId, redirectUrl, scope, authBaseUrl, clientSecret, requestTokenBaseUrl = string.Empty;
-    
-    public IndexModel(ILogger<IndexModel> logger, IConfiguration configuration)
+    private readonly ILogger<IndexModel> logger;
+    private readonly IConfiguration configuration;
+    private readonly HttpClient httpClient;
+    private readonly string clientId, redirectUrl, scope, authBaseUrl, clientSecret, requestTokenBaseUrl;
+    private readonly UserCalendarConfiguration userCalendarConfig;
+
+    public IndexModel(ILogger<IndexModel> logger, IConfiguration configuration, HttpClient httpClient)
     {
-        _logger = logger;
-        _configuration = configuration;
-
-        authBaseUrl = _configuration["GoogleAPI:AuthBaseUrl"];
-        clientId = _configuration["GoogleAPI:ClientID"];
-        redirectUrl = _configuration["GoogleAPI:RedirectUrl"];
-        scope = _configuration["GoogleAPI:Scope"];
-        clientSecret = _configuration["GoogleAPI:Secrets"];
-        requestTokenBaseUrl = _configuration["GoogleAPI:RequestTokenUrl"];
+        this.logger = logger;
+        this.configuration = configuration;
+        this.httpClient = httpClient;
+
+        authBaseUrl = this.configuration["GoogleApi:AuthBaseUrl"];
+        clientId = this.configuration["GoogleApi:ClientID"];
+        redirectUrl = this.configuration["GoogleApi:RedirectUrl"];
+        scope = this.configuration["GoogleApi:Scope"];
+        clientSecret = this.configuration["GoogleApi:Secret"];
+        requestTokenBaseUrl = this.configuration["GoogleApi:RequestTokenUrl"];
+        userCalendarConfig = UserCalendarConfiguration.Instance;
+        IsConnected = !string.IsNullOrEmpty(userCalendarConfig.RefreshToken);
+        IsWorkHourSet = !string.IsNullOrEmpty(userCalendarConfig.OpeningTime);
     }
 
-    [BindProperty]
-    public List<string> CheckedDays {get; set;}
+    [BindProperty] public List<string> CheckedDays { get; set; }
 
-    [BindProperty]
-    public List<string> AllDays {get; set;} = new List<string>
+    public string[] AllDays { get; } =
     {
-            "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"
+        "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"
     };
 
-    public bool IsConnected {get; set;} =  !string.IsNullOrEmpty(AppConfig.GetUserConfig().RefreshToken);
-    public bool IsWorkHourSet {get; set;} = !string.IsNullOrEmpty(AppConfig.GetUserConfig().OpeningTime); 
+    public bool IsConnected { get; set; }
+    public bool IsWorkHourSet { get; set; }
+    public string RedirectLink { get; set; }
 
     public async Task OnGet(string code)
     {
-        if (string.IsNullOrEmpty(code))
+        RedirectLink = authBaseUrl +

```

Comment:
Always set the link so that the user can re-authenticate even when they already did.